player/lua/input.lua

```diff
@@ -0,0 +1,69 @@
+--[[
+This file is part of mpv.
+
+mpv is free software; you can redistribute it and/or
+modify it under the terms of the GNU Lesser General Public
+License as published by the Free Software Foundation; either
+version 2.1 of the License, or (at your option) any later version.
+
+mpv is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public
+License along with mpv.  If not, see <http://www.gnu.org/licenses/>.
+]]
+
+local utils = require "mp.utils"
+local input = {}
+
+function input.get(t)
+    local console_enabled = mp.get_property_native("user-data/console/enabled")
+    if console_enabled == nil then
+        return nil, "The console script is not loaded."
+    end
+    if console_enabled then
+        return nil, "The console is already open."
+    end
```

Comment:
Could this cause race conditions? What if two scripts were to run this at the same time? If `console.lua` failed to update the `user-data` property in time then it would enable one of the inputs, and just discard the other. The script that triggered the discarded input would nonetheless think it is open. Could this lead to unexpected behaviour?

One way to prevent any race conditions would be to have `console.lua` not just silently return when receiving an input request while open, but to have it send some sort of message back to the script informing it that it could not open. Even just sending a `mp.commandv('script-message-to', input_caller, 'input-event', 'closed', '')` would probably work. The downside of this is that you don't get the convenient `nil, error` return from the `input.get` call. However, given the asynchronous nature of this whole approach I'm not sure it's possible to have only synchronous error messages. It may be worth abandoning the `console_enabled` test and just documenting that a specific callback will be called if the input could not be opened for consistency. Whether that should be `closed`, or some dedicated callback, I'm not sure.

Reply:
Completely agreed. I made it return an error because avih requested it 2 years ago, but I find it useless. It has to add code to update `user-data/console/enabled` (which also bloats every `--log-file` with 2 extra lines), it can't be checked reliably without race conditions, and I don't see why you would need to check for an error if everything is done through callbacks, it is like the return value `mp.command_native_async()` or Javascript promises. I'm all for removing this if maintainers are fine with it.

Reply:
You can remove it imo

Reply:
It's gone. 🎉